src/actions/channels.ts

```diff
@@ -724,6 +724,27 @@ export function deleteChannel(channelId: string): ActionFunc {
     };
 }
 
+export function unarchiveChannel(channelId: string): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let state = getState();
+
+        try {
+            await Client4.unarchiveChannel(channelId);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        state = getState();
+        const viewArchivedChannels = state.entities.general.config.ExperimentalViewArchivedChannels === 'true';
```

Comment:
Can we use the selector to make it less error prone and easier to read?  Something like what you did in other place.
```javascript
        const state = getState();
        const config = getConfig(state);
        const viewArchivedChannels = config.ExperimentalViewArchivedChannels === 'true';
```

Reply:
hmmm. websockets had a method called getConfig(), this does not.

Reply:
good catch